spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/src/main/java/org/springframework/cloud/function/adapter/gcloud/GcfHandler.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.function.adapter.gcloud;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.cloud.functions.Context;
+import com.google.cloud.functions.HttpResponse;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+
+import org.springframework.cloud.function.context.AbstractSpringFunctionAdapterInitializer;
+import org.springframework.messaging.Message;
+
+public class GcfHandler<O> extends AbstractSpringFunctionAdapterInitializer<Context> {
```

Comment:
Would it be better to just have the background function and httpFunction extend from the AbstractSpringFunctionAdapterInitializer directly? I think right now for HttpFunction to extend from GcfHandler might not be ideal if it doesn't have access to the Context.